src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/relationship.effects.ts

```diff
@@ -119,17 +158,36 @@ export class RelationshipEffects {
               return this.relationshipService.addRelationship(type.id, item1, item2, undefined, nameVariant);
             }
           }
-        )
-      ).pipe(take(1))
-      .subscribe();
+        ),
+        take(1),
+        switchMap(() => this.refreshWorkspaceItemInCache(submissionId)),
+      ).subscribe((submissionObject: SubmissionObject) => this.store.dispatch(new SaveSubmissionSectionFormSuccessAction(submissionId, [submissionObject], false)));
   }
 
-  private removeRelationship(item1: Item, item2: Item, relationshipType: string) {
+  private removeRelationship(item1: Item, item2: Item, relationshipType: string, submissionId: string) {
     this.relationshipService.getRelationshipByItemsAndLabel(item1, item2, relationshipType).pipe(
-      take(1),
-      hasValueOperator(),
       mergeMap((relationship: Relationship) => this.relationshipService.deleteRelationship(relationship.id, 'none')),
-      take(1)
-    ).subscribe();
+      take(1),
+      switchMap(() => this.refreshWorkspaceItemInCache(submissionId)),
+    ).subscribe((submissionObject: SubmissionObject) => {
+      this.store.dispatch(new SaveSubmissionSectionFormSuccessAction(submissionId, [submissionObject], false))
+    });
+  }
+
+  refreshWorkspaceItemInCache(submissionId: string): Observable<SubmissionObject> {
```

Comment:
Please add TypeDocs to describe this new method.

Reply:
Added TypeDoc and made method private